tripy/tripy/function_registry.py

```diff
@@ -98,7 +98,26 @@ def _get_annotations(self):
     def matches_arg_types(self, args, kwargs) -> "Result":
         from tripy.utils.result import Result
 
-        def matches_type(name: str, annotation: type, arg: Any):
+        def sanitize_name(annotation):
+            # typing module annotations are likely to be better when pretty-printed due to including subscripts
+            return annotation if annotation.__module__ == "typing" else annotation.__qualname__
+
+        def render_arg_type(arg: Any) -> str:
+            # it is more useful to report more detailed types for sequences/tuples in error messages
+            from typing import List, Tuple
+
+            if isinstance(arg, List):
+                if len(arg) == 0:
+                    return "List"
+                return f"List[{render_arg_type(arg[0])}]"
+            if isinstance(arg, Tuple):
+                return f"Tuple[{', '.join(map(render_arg_type, arg))}]"
```

Comment:
Can this logic be generalized to all sequences? Something like:
```py
if isinstance(arg, Sequence):
    return f"{type(arg).title()}[{', '.join(map, render_arg_type, arg)}]"
```
Then we don't need special logic for lists either.

Reply:
I can do it that way. I had them separately there because I think it might make sense not to have the brackets for an empty list (i.e., the type of the contents is unknown), but it may not be an important distinction.

Reply:
There is no `title()` method on many type instances, I noticed.

Reply:
Oops, meant to do `type(arg).__name__.title()`